src/traces/funnel/style.js

```diff
@@ -0,0 +1,60 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var d3 = require('d3');
+
+var Drawing = require('../../components/drawing');
+var Color = require('../../components/color');
+
+var styleTextPoints = require('../bar/style').styleTextPoints;
+
+function style(gd, cd) {
+    var s = cd ? cd[0].node3 : d3.select(gd).selectAll('g.funnellayer').selectAll('g.trace');
+
+    s.style('opacity', function(d) { return d[0].trace.opacity; });
+
+    s.each(function(d) {
+        var gTrace = d3.select(this);
+        var trace = d[0].trace;
+
+        gTrace.selectAll('.point > path').each(function(di) {
+            if(!di.isBlank) {
+                var cont = trace.marker;
+
+                d3.select(this)
+                    .call(Color.fill, di.mc || cont.color)
+                    .call(Color.stroke, di.mlc || cont.line.color)
+                    .call(Drawing.dashLine, cont.line.dash, di.mlw || cont.line.width)
+                    .style('opacity', trace.selectedpoints && !di.selected ? 0.3 : 1);
```

Comment:
For bonus points, as `funnel` have (only) one `marker` style container, it should be relatively easy to implement `[un]selected.marker` styles like bar traces have:

https://github.com/plotly/plotly.js/blob/adf9bee14a8badcb5b2f49065fd594623421c82e/src/traces/bar/attributes.js#L199-L216

Maybe you can come back to when `funnelarea` is completed.